projects/optic/src/commands/bundle/bundle.ts

```diff
@@ -599,6 +626,49 @@ function bundleMatchingRefsAsComponents<T>(
     }
   }, specCopy);
 
+  if (options.rewriteMapping) {
+    let $refs = await $RefParser.resolve(sourcemap.rootFilePath);
+
+    const refKeys = $refs.paths();
+
+    // handle mapping
+    const mappingPattern = [
```

Comment:
1. find mapping within schemas
2. associate the $ref with the resolved ref graph 
3. if it exists apply a patch. It should exist, but it's also valid to use a URL or other pointer type here.